File: swift/tests/TestCommons.swift
import Foundation

@main
struct TestCommons {
    static func main() {
        let data = Data("alpha\nbeta\n".utf8)
        let src = "/tmp/swift-commons-src.txt"
        let dst = "/tmp/swift-commons-dst.txt"
        do {
            try data.write(to: URL(fileURLWithPath: src))
            FileManager.default.createFile(atPath: dst, contents: nil)
            let out = try FileHandle(forWritingTo: URL(fileURLWithPath: dst))
            defer { try? out.close() }
            try commons.copyFile(src, out)
            let got = try Data(contentsOf: URL(fileURLWithPath: dst))
            if got != data { exit(1) }
            try? FileManager.default.removeItem(atPath: src)
            try? FileManager.default.removeItem(atPath: dst)
            exit(0)
        } catch {
            exit(1)
        }
    }
}

File: elixir/tests/test_commons.exs
defmodule TestCommons do
  use ExUnit.Case, async: true

  test "write and read roundtrip" do
    tmp = Path.join(System.tmp_dir!(), "commons-test-input.txt")
    content = "alpha\nbeta\n"
    :ok = File.write(tmp, content)
    assert File.read!(tmp) == content
    :ok = File.rm(tmp)
  end
end

ExUnit.start()
